src/cmap/events.ts

```diff
@@ -312,41 +311,59 @@ const extractCommand = (command: any) => {
   }
 
   if (command.query && command.query.$query) {
-    let result: any;
-    if (command.ns === 'admin.$cmd') {
-      // upconvert legacy command
+    let result: Document;
+    const commandObject: any = command;
```

Comment:
:thinking: it looks to me like more refactoring is needed to ensure proper typing, otherwise we might as well leave `command: any` above and be done with it today

Reply:
Fixed this, found a typo!
`returnFieldsSelector` is mapped to `projection` should be `returnFieldSelector` right?

Reply:
nice! yes, that's correct.